### PR TITLE
[XR] add a new function to layers feature

### PR DESCRIPTION
**Pull request by RaananW**
There was no way of tracking existing layers and adding a new one.

### CODE REVIEW

**Review by sebavan — APPROVED** [review]

File: src/XR/features/WebXRLayers.ts [hunk 1]
```diff
@@ -242,6 +257,15 @@ export class WebXRLayers extends WebXRAbstractFeature {
         return new WebXRProjectionLayerWrapper(projLayer, this._xrWebGLBinding);
     }
 
+    /**
+     * Add a new layer to the already-existing list of layers
+     * @param wrappedLayer the new layer to add to the existing ones
+     */
+    public addXRSessionLayer(wrappedLayer: WebXRLayerWrapper) {
```
Comment (rgerd) [inline src/XR/features/WebXRLayers.ts:264]: Should there also be a `getXRSessionLayers`?

File: src/XR/features/WebXRLayers.ts [hunk 2]
```diff
@@ -242,6 +257,15 @@ export class WebXRLayers extends WebXRAbstractFeature {
...
+        this._existingLayers.push(wrappedLayer);
+        this.setXRSessionLayers(this._existingLayers);
```
Comment (rgerd) [inline src/XR/features/WebXRLayers.ts:266]: Should `setXRSessionLayers` set `this._existingLayers`? 
This doesn't seem like a function that will be called very often so it feels like you could do something like `this.setXRSessionLayers([...this._existingLayers, wrappedLayer])` and then in `setXRSessionLayers`, `this._existingLayers = wrappedLayers.slice(0)`

File: src/XR/features/WebXRLayers.ts [hunk 3]
```diff
@@ -118,6 +119,7 @@ class WebXRCompositionLayerRenderTargetTextureProvider extends WebXRLayerRenderT
  * @hidden
  */
 export class WebXRProjectionLayerWrapper extends WebXRCompositionLayerWrapper {
+    public babylonLayer: Nullable<Layer>;
```
Comment (rgerd) [inline src/XR/features/WebXRLayers.ts:122]: I didn't even realize we had a Layer type! 
Was this an intentional change? I don't see it being used elsewhere...

Comment (RaananW) [reply src/XR/features/WebXRLayers.ts:122]: oh! this is a preparation for a further change I was planning, good catch. I will remove it from here for now.

File: src/XR/features/WebXRLayers.ts (hunk 2 above)
Comment (RaananW) [reply src/XR/features/WebXRLayers.ts:266]: I thought about it. At the beginning I exposed the existing layers, and then I thought the flow is - create a wrapper (and have a reference to it) and then add it to the session. I already have the wrapper (and am using it somewhere else), I don't know when I will need the entire array. Apart from when I want to remove the "base" layer, but that would stop rendering. What do you think?

File: src/XR/features/WebXRLayers.ts (hunk 3 above)
Comment (RaananW) [reply src/XR/features/WebXRLayers.ts:122]:
```suggestion
```

File: src/XR/features/WebXRLayers.ts (hunk 2 above)
Comment (RaananW) [reply src/XR/features/WebXRLayers.ts:265]: (BTW - I resolved the other conversation. I keep this one open for further discussion on this subject)

**Pull request merged**